src/Integration.Vsix/CFamily/ActiveCompilationDatabaseTracker.cs

```diff
@@ -0,0 +1,133 @@
+ */
+
+using System.ComponentModel.Composition;
+using Microsoft.VisualStudio.Threading;
+using SonarLint.VisualStudio.CFamily;
+using SonarLint.VisualStudio.CFamily.CMake;
+using SonarLint.VisualStudio.Core;
+using SonarLint.VisualStudio.Core.CFamily;
+using SonarLint.VisualStudio.Core.ConfigurationScope;
+using SonarLint.VisualStudio.Core.Initialization;
+using SonarLint.VisualStudio.Core.Synchronization;
+using SonarLint.VisualStudio.Infrastructure.VS.Initialization;
+using SonarLint.VisualStudio.SLCore.Core;
+using SonarLint.VisualStudio.SLCore.Service.Analysis;
+
+namespace SonarLint.VisualStudio.Integration.Vsix.CFamily;
+
+[Export(typeof(IActiveCompilationDatabaseTracker))]
+[PartCreationPolicy(CreationPolicy.Shared)]
+internal sealed class ActiveCompilationDatabaseTracker : IActiveCompilationDatabaseTracker
+{
+    private readonly ICMakeCompilationDatabaseLocator cMakeCompilationDatabaseLocator;
+    private readonly IActiveVcxCompilationDatabase activeVcxCompilationDatabase;
+    private readonly IActiveConfigScopeTracker activeConfigScopeTracker;
+    private readonly ISLCoreServiceProvider serviceProvider;
+    private readonly IThreadHandling threadHandling;
+    private readonly IAsyncLock asyncLock;
+    private bool isDisposed;
+    private string lastConfigScopeId;
+
+    public IInitializationProcessor InitializationProcessor { get; }
+    public string DatabasePath { get; private set; }
+
+    [ImportingConstructor]
+    public ActiveCompilationDatabaseTracker(
+        ICMakeCompilationDatabaseLocator cMakeCompilationDatabaseLocator,
+        IActiveVcxCompilationDatabase activeVcxCompilationDatabase,
+        IActiveConfigScopeTracker activeConfigScopeTracker,
+        IInitializationProcessorFactory initializationProcessorFactory,
+        IAsyncLockFactory asyncLockFactory,
+        IThreadHandling threadHandling,
+        ISLCoreServiceProvider serviceProvider)
+    {
+        this.cMakeCompilationDatabaseLocator = cMakeCompilationDatabaseLocator;
+        this.activeVcxCompilationDatabase = activeVcxCompilationDatabase;
+        this.activeConfigScopeTracker = activeConfigScopeTracker;
+        this.threadHandling = threadHandling;
+        this.serviceProvider = serviceProvider;
+        asyncLock = asyncLockFactory.Create();
+        InitializationProcessor = initializationProcessorFactory.CreateAndStart<ActiveCompilationDatabaseTracker>([],
+            async () =>
+            {
+                if (isDisposed)
+                {
+                    return;
+                }
+                await activeVcxCompilationDatabase.EnsureDatabaseInitializedAsync();
+                activeConfigScopeTracker.CurrentConfigurationScopeChanged += ActiveConfigScopeTrackerOnCurrentConfigurationScopeChanged;
+                await HandleConfigScopeEventAsync();
+            });
+    }
+
+    private void ActiveConfigScopeTrackerOnCurrentConfigurationScopeChanged(object sender, EventArgs e) =>
```

Comment:
The name is very long and adding the event in the name doesn't really bring any more info

```suggestion
    private void OnCurrentConfigurationScopeChanged(object sender, EventArgs e) =>
```